docs/tutorials/csharp/stable-diffusion-csharp.md

```diff
@@ -0,0 +1,331 @@
+
+
+## Contents
+{: .no_toc }
+
+* TOC placeholder
+{:toc}
+
+## Prerequisites
+This tutorial can be run locally or in the cloud by leveraging Azure Machine Learning compute.
+
+- [Download the Source Code from GitHub](https://github.com/cassiebreviu/StableDiffusion)
+
+To run locally:
+
+- [Visual Studio](https://visualstudio.microsoft.com/downloads/) or [VS Code](https://code.visualstudio.com/Download)
+- A GPU enabled machine with CUDA EP Configured. This was built on a GTX 3070 and it has not been tested on anything smaller. Follow [this tutorial to configure CUDA and cuDNN for GPU with ONNX Runtime and C# on Windows 11](https://onnxruntime.ai/docs/tutorials/csharp/csharp-gpu.html)
+
+To run in the cloud with Azure Machine Learning:
+
+- [Azure Subscription](https://azure.microsoft.com/free/)
+- [Azure Machine Learning Resource](https://azure.microsoft.com/services/machine-learning/)
+
+## Use Hugging Face to download the Stable Diffusion models
+
+The Hugging Face site has a great library of open source models. We will leverage and download the [ONNX Stable Diffusion models from Hugging Face](https://huggingface.co/models?sort=downloads&search=Stable+Diffusion).
+
+ - [ONNX Stable Diffusion Models for v1.4](https://huggingface.co/CompVis/stable-diffusion-v1-4/tree/onnx)
+ - [ONNX Stable Diffusion Models for v1.5](https://huggingface.co/runwayml/stable-diffusion-v1-5/tree/onnx)
+
+
+Once you have selected a model version repo, click `Files and Versions`, then select the `ONNX` branch. If there isn't an ONNX model branch available, use the `main` branch and convert it to ONNX. See the [ONNX conversion tutorial for PyTorch](https://learn.microsoft.com/windows/ai/windows-ml/tutorials/pytorch-convert-model) for more information.
+
+- Clone the repo:
+```text
+git lfs install
+git clone https://huggingface.co/CompVis/stable-diffusion-v1-4 -b onnx
+```
+
+- Copy the folders with the ONNX files to the C# project folder `\StableDiffusion\StableDiffusion`. The folders to copy are: `unet`, `vae_decoder`, `text_encoder`, `safety_checker`. 
+
+## Inference with C#
+Now lets start to breakdown how to inference in C#! The `unet` model takes the text embedding of the user prompt created by the [CLIP model](https://huggingface.co/docs/transformers/model_doc/clip) that connects text and image. The latent noisy image is created as a starting point. The scheduler algorithm and the `unet` model work together to denoise the image to create an image that represents the text prompt. Lets look at the code.
+
+## Main Function
+The main function sets the prompt, number of inference steps, and the guidance scale. It then calls the `UNet.Inference` function to run the inference.
+
+The properties that need to be set are:
+ - `prompt` - The text prompt to use for the image
+ - `num_inference_steps` - The number of steps to run inference for
```

Comment:
These could have a little more explanation

Reply:
ie does the number of inference steps improve the quality of the model?

Reply:
added details and note for hugging face blog to learn more